### PR TITLE
Add the -dunique-ids and -dno-unique-ids options to the compilers

### PR DESCRIPTION
**Pull request by shindere**
These options allow to control whether identifiers are made unique by
appending a stamp to them when dumping intermediate representations or not.

The default is to print the stamp, as is done currently.

The "-no-unique-ids" option is useful e.g. to simplify the comparison
between a produced intermediate reprsentation (-dlambda, say) and the
expected one, in the context of the testsuite, for instance.

This is a follow-up to a discussion started on caml-devel. On the list,
@damiendoligez suggested "making it a variant of -dlambda (for example,
-dlambda-unnumbered)" (because the original motivation is to simplify the
tests on the lambda code), but then it seemed to me the implementation would
be both simpler and more straightforward if the feature of making
identifiers unique would be kept orthogonal to the intermediate
reprresentation.

### CODE REVIEW

**Comment by lpw25** [comment]
Since this only affects `-d...` options might I suggest `-dunique-ids` and `-dno-unique-ids` to make it more obvious that it is not for normal users.

**Comment by shindere** [comment]
Leo White (2018/02/19 13:59 +0000):
> Since this only affects `-d...` options might I suggest `-dunique-ids`
> and `-dno-unique-ids` to make it more obvious that it is not for
> normal users.

Sure I could do that. Are we sure, though, that this won't affect
anything else one day?

**Comment by shindere** [comment]
@lpw25 I updated the code to rename the options according to your suggestion.

**Review by damiendoligez — APPROVED** [review]
Looks good except for an indentation change that I suggested directly to Sebastien.

**Comment by shindere** [comment]
Rebased, fixed indentation and merged.

Thanks a lot for the review @damiendoligez

File: Changes [hunk 1]
```diff
@@ -104,6 +104,9 @@ Working version
 - GPR#1554: warnings 52 and 57: fix reference to manual detailed explanation
   (Florian Angeletti, review by Thomas Refis and Gabriel Scherer)
 
+- GPR#1618: add the -dno-unique-ids and -dunique-ids compiler flags
+  (Sébastien Hinderer, review by xxx)
```
Comment (gasche) [inline Changes:108]: "review by xxx" would be Leo White and Damien Doligez?

**Comment by shindere** [comment]
Ah yes, sorry for having forgotten the credits and thanks for the
reminder, @gasche.
Fixed.

**Pull request merged**